### DISCUSSION TITLE
Questions eForms Implementation - Please Help

### DISCUSSION THREAD

**Discussion by JakubBalaz** [OP-TED/eForms-SDK#585] [Q&A] [answered]
During implementation of eForms, we are faced with the following questions, which are very important for us in order to proceed successfully:

1.	Concerning Change Notice: According to our information not all field can be changed via Change Notice (i.e. main classification etc.). In the 5th TED technical workshop it was announced that a list of fields that can be corrected will be provided. Can you let us know when we can expect the list or can you tell us, where to find the list?
2.	Concerning Revocation Notice (cancellation of the tender): We are not sure which form to use for this. Can CAN 29-32 be used, or should we use the change notice for this?

3.	Field BT-3202-Contract (Contract Tender Indentifier): Since we are not using SDK for implementation, we would like to leave this field out within our framework. However, we are not sure if we can leave this field out, or if we will get trouble with validation, because we do not understand the corresponding rule:

BT-3202-Contract is mandatory for a notice with subtype 29 under the condition: Contract Technical ID (OPT-316-Contract) does not exist or there is no Tender (OPT-321-Tender)

What is Contract Technical ID (OPT-316-Contract)? AND why should it be mandatory to provide a Contract Tender Identifier if there is no tender? This rule does not appear logical in our view. Please advice. Can we eliminate this field within our framework?

4.	The SDK rules according to the translation table appear contradictory in several instances. Here are a few Examples:
a)	Form 29: Tender identifier reference (OPT-320-LotResult) and ID – Contract (OPT-315-LotResult)  (Form 29): the rule states that these fields are MANDATORY, if there ist NO Lot Result or if there is no Tender etc…..

"OPT-320-LotResult is mandatory for a notice with subtype 29 under the condition: there is no LotResult, or there is no Tender for the Lot covered by the LotResult and no Tender for any Group of Lot that would contain the Lot of the LotResult, or there is no tender and no lot is specified for the LotResult, or the competition is ongoing“.

Why should these fields be mandatory, if there is no Lot result or there is no tender or the competition is still ongoing? In those cases it would make sense, to eliminate these fields. In our opinion it would only make sense to Link to the tender ID and contact ID, if there is a LotResult and the competition is closed. 

b)	BT-171-Tender (Rank in the list of winners) is a mandatory field, IF BT-1711-Tender (The tender was ranked) is set to “FALSE”. Why should a user be forced to provide a rank in the list of winners if there is no ranking in the first place. For us this rule does not appear logical.

 
Please advice, if those rules will be corrected in future SDK-Versions.

Thanks a lot for your help.

**Comment by YvesJo** [comment, marked as answer]
Hi,
Regarding non technical questions, please get directly in touch with the helpdesk: [ted@publications.europa.eu](mailto:ted@publications.europa.eu).
The rule statement for OPT-320-LotResult is incorrect and belongs to a set of rules for which Conditionally Mandatory and Conditionally Forbidden rules are sharing a same condition (condition that shall be verified for one and not verified for the other). The labels for these rules are being updated.
